### DISCUSSION TITLE
Is there a way to pass the full file path or the repo directory when using openCommand

### DISCUSSION THREAD

**Discussion by donovan** [jesseduffield/lazygit#1961] [Q&A]
Is there a way to pass the full file path or the repo directory when using `openCommand`?

My use case is the following:

In neovim 0.7 you can do the following:

```
nvim --listen /tmp/nvim.sock
```

In lazygit config I can add the following:
```
os:
    openCommand: 'nvim --server /tmp/nvim.sock --remote {{filename}}'
```

This will open the file in the neovim instance listening on the socket, ready for editing when you press `o`.

This works great but I would like to be able to run a nvim instance per repo rather than a global instance where I send all files. In order to achieve this I would to use a small wrapper script but I need to work out how to get lazygit to send the full file path or the repo directory path. The templated `filename` variable is relative to the repo, therefore I cant tell what repo the file resides in in my wrapper script.

This is what I am after:
```
os:
    openCommand: 'nsend {{filename}} {{repo}}'
```

or
```
os:
    openCommand: 'nsend {{filepath}}'
```

and my wrapper would look like this:
```
#!/bin/bash

file_path="$1"
repo="$2"

nvim --server "/tmp/nvim.${repo}.sock" --remote "${file_path}"
```
The repo dir would be preferable as if I just get the filepath I need to mess with basename/dirname and git rev-parse --show-toplevel
I looked at the wiki and most of the docs and could not work out how to do it.  Is this possible? If not would the developers consider adding this template var?

Thanks

**Comment by Ryooooooga** [comment]
You can write any bash script in `openCommand`.

```yaml
os:
    openCommand: open "$(git rev-parse --show-toplevel)/"{{filename}}
```
